### ISSUE TITLE
Bind not Successfull

### ISSUE THREAD

**Issue by jftripp** [heiglandreas/authLdap#100]
So, I'm using an LDAP URI based on the convention in the plugin. I have confirmed that this URI works fine in an LDAP browser, but it doesn't work in the plugin. This is the debug info:

[23-Jan-2016 21:38:03 UTC] [AuthLDAP] LDAP authentication failed with exception: bind was not successfull
[23-Jan-2016 21:38:02 UTC] [AuthLDAP] connect to LDAP server
[23-Jan-2016 21:38:02 UTC] [AuthLDAP] about to do LDAP authentication
[23-Jan-2016 21:38:02 UTC] [AuthLDAP] User 'mollytripp@gmail.com' logging in

Not sure how to debug this.

**Comment by heiglandreas** [comment]
Hi @jftripp. Without having dug deeper I've got a question: Have you confirmed that the URI works in an LDAP-Browser *on the webserver* or *on your local machine*? Background is, that the network-structure often doesn't allow access to the LDAP-Server from external webservers. So it works fine from your local machine (inside the network) but not from the webserver (in the DMZ or outside the internal network). You can test that if you have access to a terminal on the webserver by trying to connect to that LDAP-URI.

If that *does* work, I will need a bit more information like the LDAP-URI, what LDAP-Server you are trying to connect to and whether anonymous binding is used and what filters you are using. But we'l come to that later :wink:

Cheers

**Comment by jftripp** [comment]
Well, I know that the LDAP is allowing connections from the server because other plugins are binding. However, I’m not sure how those plugins are generating the URI, because the plugin does that from input fields on the form.

This is the URI that I’m using:

ldaps://CN=LDAPUser,O=Darwino:LDAPUserP00@playground.darwino.com:636/O=Darwino

I’ve also tried

ldaps://LDAPUser:LDAPUserP00@playground.darwino.com:636/O=Darwino

and

ldaps://CN=LDAPUser,O=Darwino@playground.darwino.com:636/O=Darwino

And

ldaps://CN=LDAPUser,O=Darwino:LDAPUserP00@playground.darwino.com:636



On Jan 24, 2016, at 2:23 AM, Andreas Heigl <notifications@github.com> wrote:


Hi @jftripp<https://github.com/jftripp>. Without having dug deeper I've got a question: Have you confirmed that the URI works in an LDAP-Browser on the webserver or on your local machine? Background is, that the network-structure often doesn't allow access to the LDAP-Server from external webservers. So it works fine from your local machine (inside the network) but not from the webserver (in the DMZ or outside the internal network). You can test that if you have access to a terminal on the webserver by trying to connect to that LDAP-URI.

If that does work, I will need a bit more information like the LDAP-URI, what LDAP-Server you are trying to connect to and whether anonymous binding is used and what filters you are using. But we'l come to that later [:wink:]

Cheers

—
Reply to this email directly or view it on GitHub<https://github.com/heiglandreas/authLdap/issues/100#issuecomment-174266720>.

Dr. John F. Tripp
Assistant Professor
Information Systems Department
Hankamer School of Business
One Bear Place #98005
Baylor University, Waco, TX 76798-8005
(254) 710-4097 (w)
(254) 292-8417 (c)

**Comment by heiglandreas** [comment]
That LDAP-URI you are using *should* work. As you are using the default port for LDAPS would you mind ommiting the Port and use ```ldaps://CN=LDAPUser,O=Darwino:LDAPUserP00@playground.darwino.com/O=Darwino``` instead to see whether it works? That would then mean I have to check the URI-deconstruction, but I'll then take care of that....

**Comment by jftripp** [comment]
This is not working…too bad - I also found another plugin that doesn’t work. I’m doing some debugging to see what is the difference between the plugins that DO work, and the plugins that don’t.

On Jan 24, 2016, at 10:37 AM, Andreas Heigl <notifications@github.com<mailto:notifications@github.com>> wrote:


That LDAP-URI you are using should work. As you are using the default port for LDAPS would you mind ommiting the Port and use ldaps://CN=LDAPUser,O=Darwino:LDAPUserP00@playground.darwino.com/O=Darwino instead to see whether it works? That would then mean I have to check the URI-deconstruction, but I'll then take care of that....

—
Reply to this email directly or view it on GitHub<https://github.com/heiglandreas/authLdap/issues/100#issuecomment-174314568>.

Dr. John F. Tripp
Assistant Professor
Information Systems Department
Hankamer School of Business
One Bear Place #98005
Baylor University, Waco, TX 76798-8005
(254) 710-4097 (w)
(254) 292-8417 (c)

**Comment by heiglandreas** [comment]
In ldap.php replace line 154 with the following:

```php
            throw new AuthLDAP_Exception('bind was not successfull: ' . ldap_error($this->_ch));
```

That should give the exact error-code from the LDAP-Server in the debug-messages.

I'll have to add that to the code! Thanks for raising this issue!

**Comment by heiglandreas** [comment]
This error-message is now included in Release 1.4.15
